common/tools/dev-tool/README.md

```diff
@@ -0,0 +1,223 @@
+    - `prep` (prepare samples for local source-linked execution)
+    - `run` (execute a sample or all samples within a directory)
+
+The `dev-tool about` command will print some information about how to use the command. All commands additionally accept the `--help` argument, which will print information about the usage of that specific command. For example, to show help information for the `resolve` command above, issue the command `dev-tool package resolve --help`.
+
+## Extending the Tool
+
+The source hierarchy matches the command hierarchy. Every sub-command has its own folder and `index.ts` file in `src/commands`, where `src/commands/index.ts` defines the behavior of the root `dev-tool` command, and each subfolder's `index.ts` file describes a nested sub-command. Every leaf node in the command tree ("leaf command") has its own TypeScript file. For example, `src/commands/about.ts` defines the behavior of the `dev-tool about` command, and `src/commands/package/resolve.ts` defines the behavior of the `dev-tool package resolve` command.
+
+### Command Interface
+
+Every command file's exports must implement the `CommandModule` interface defined in `src/util/commandModule.ts`. The interface requires that every command export a constant `commandInfo` that implements the `CommandInfo` interface defined in the same file. The `CommandInfo` interface specifies the name, description, and options (command-line arguments) of the command. The command module must also export an async handler function as its default export. Two helper functions, `leafCommand` and `subCommand` are provided to assist with development and to provide strong type-checking when
+extending dev-tool.
+
+### Creating a new leaf command
+
+To create a new leaf command in one of the existing sub-command, create a new TypeScript file for that command. Make sure that your module exports the required `commandInfo` and default handler function. When creating a command, use the `leafCommand` helper to get a strongly-typed `options` parameter for your handler.
+
+As an example, we can create a new `hello-world` command under the `dev-tool package` sub-command. The command will print out a string using the many different logging functions. It will accept an argument `--echo <string here>` that specifies the string to be printed.
+
+`src/commands/package/hello-world.ts`
+```typescript
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license
+
+import { createPrinter } from "../../util/printer";
+import { leafCommand } from "../../util/commandBuilder";
+
+const log = createPrinter("hello-world");
+
+export const commandInfo = {
+  name: "hello-world",
+  description:
+    "print a lovely message",
+  options: {
+      echo: {
+          kind: "string",
+          description: "override the message to be printed",
+          default: "Hello world!"
+      }
+  }
+} as const;
+
+export default leafCommand(commandInfo, async (options) => {
```

Comment:
maybe for clarity you could assign the result of leafCommand to a constant and export that as default?

Reply:
If that's the pattern that we prefer, then I am happy to make that change. I have no strong feelings about it but I do tend to prefer the `export default <value>` rather than `const <name> = <value>; export default <name>;` style. I feel that it's just a bit more declarative.

Reply:
I'm fine with this